dask/dataframe/tests/test_io.py

```diff
@@ -630,16 +633,20 @@ def test_to_castra():
     finally:
         c.drop()
 
-    dsk, keys = a.to_castra(compute=False)
-    assert isinstance(dsk, dict)
-    assert isinstance(keys, list)
-    c, last = keys
-    assert last[1] == a.npartitions - 1
+    delayed = a.to_castra(compute=False)
+    assert isinstance(delayed.dask, dict)
+    c, _ = delayed.compute()
```

Comment:
What's returned here? I'd expect

```python
c1 = a.to_castra(compute=True)
c2 = a.to_castra(compute=False).compute()
tm.assert_frame_equal(c1[:], c2[:])
```

to pass. From glancing at the code in the referenced commit, it looks like this is not the case. Unless there is a good reason for the current behavior, would you mind adding a fix to make these equivalent?